tests/core/test_core.py

```diff
@@ -578,35 +599,94 @@ def mock_typing() -> None:
         Thread(controller.show_typing_notification()).start()
 
         if active_conversation_info:
-            set_footer_text.assert_has_calls(
-                [
-                    mocker.call(
-                        [
-                            ("footer_contrast", "hamlet "),
-                            ("footer", " is typing"),
-                        ]
-                    ),
-                    mocker.call(
-                        [
-                            ("footer_contrast", "hamlet "),
-                            ("footer", " is typing."),
-                        ]
-                    ),
-                    mocker.call(
-                        [
-                            ("footer_contrast", "hamlet "),
-                            ("footer", " is typing.."),
-                        ]
-                    ),
-                    mocker.call(
-                        [
-                            ("footer_contrast", "hamlet "),
-                            ("footer", " is typing..."),
-                        ]
-                    ),
-                ]
-            )
-            set_footer_text.assert_called_with()
+            if length == 1:
+                set_footer_text.assert_has_calls(
+                    [
+                        mocker.call(
+                            [
+                                ("footer_contrast", f"{active_conversation_info} "),
+                                ("footer", " is typing"),
+                            ]
+                        ),
+                        mocker.call(
+                            [
+                                ("footer_contrast", f"{active_conversation_info} "),
+                                ("footer", " is typing."),
+                            ]
+                        ),
+                        mocker.call(
+                            [
+                                ("footer_contrast", f"{active_conversation_info} "),
+                                ("footer", " is typing.."),
+                            ]
+                        ),
+                        mocker.call(
+                            [
+                                ("footer_contrast", f"{active_conversation_info} "),
+                                ("footer", " is typing..."),
+                            ]
+                        ),
+                    ]
+                )
+            elif length < 4:
```

Comment:
This conditional should not be present in the test. Instead add a different parameter to the test for what you expect in each case.